apiserver/pkg/util/service.go

```diff
@@ -68,13 +72,18 @@ func buildRayServiceSpec(apiService *api.RayService, computeTemplateMap map[stri
 		}
 		serveConfigSpecs = append(serveConfigSpecs, serveConfigSpec)
 	}
+	newRayClusterSpec := *buildRayClusterSpec(rayServiceDefaultVersion, nil, apiService.ClusterSpec, computeTemplateMap)
+	newRayClusterSpec.HeadGroupSpec.Template.Spec.Containers[0].Ports = append(newRayClusterSpec.HeadGroupSpec.Template.Spec.Containers[0].Ports, v1.ContainerPort{
+		Name:          defaultServePortName,
+		ContainerPort: defaultServePort,
```

Comment:
I am not familiar with the API server. Are there any use cases in that users want to use another port (e.g. 9000 in #570)?

Reply:
The RayCluster controller's behavior includes the serve port in the head service by default.

For consistency with the RayCluster controller's behavior, I think we should include this with the rest of the ports in `buildRayClusterSpec`.
Alternatively, **don't include the ports in buildRayClusterSpec**. The RayCluster controller will configure them automatically.
https://github.com/ray-project/kuberay/blob/4b0f7cbaa42116cbba2e80854e70d6964196fab5/ray-operator/controllers/ray/common/service.go#L151

Reply:
@kevin85421 I think configurability of the port via the KubeRay API server is a separate issue

The point of this PR is just to make it possible to configure a working RayService with the KubeRay API server.

Reply:
@DmitriGekhtman Thank you for your feedback and it's a great discussion, I was originally thinking we should add the serve port in the `buildRayClusterSpec` because in` kuberay-apiserver` the create server api will leverage the  `buildRayClusterSpec` to populate the template of a ray service. and for now the missing of serve port will lead to an error that the rayservice cannot expose the serve. in `rayservice-controller` it will create a ray service with no serve port and we cannot access the rayservice that create by `kuberay-apiserver`.
![image](https://user-images.githubusercontent.com/38214590/203236995-1537e541-eb9e-4176-abbb-255241bef664.png)

Reply:
The things is that we may not only use the `buildRayClusterSpec` in building a ray service, but also in creating jobs:
https://github.com/ray-project/kuberay/blob/3a512da1de184476cfdb8091bb56745ddeda93a6/apiserver/pkg/util/job.go#L22
or building ray cluster,
https://github.com/ray-project/kuberay/blob/3a512da1de184476cfdb8091bb56745ddeda93a6/apiserver/pkg/util/cluster.go#L29
in those cases, the serve port may not be necessary and it may conflict with the configuration there. So I add it in the build service layer and add a comment in `apiserver/pkg/util/cluster.go:L138`